src/Service/ProgrammesService.php

```diff
@@ -38,9 +38,17 @@ public function __construct(
         parent::__construct($repository, $mapper, $cache);
     }
 
-    public function countAllTleosByCategory(Category $category): int
+    public function countAllTleosByCategory(Category $category, $ttl = CacheInterface::NORMAL): int
     {
-        return $this->repository->countTleosByCategory($category->getDbAncestryIds(), false);
+        $key = $this->cache->keyHelper(__CLASS__, __FUNCTION__, implode('-', $category->getDbAncestryIds()), $ttl);
```

Comment:
Use $category-> getDbId(), rather than the list of all its parent ancestry. You do this a few times in this file.